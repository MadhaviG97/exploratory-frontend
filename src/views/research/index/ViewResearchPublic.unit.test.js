import React from 'react';
import ReactDOM from 'react-dom';
import ResearchPublic from './ViewResearchPublic';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
    user: { userData: {isAuth:true,first_name:'madhavi'} }
  });
jest.mock('react-router-dom', () => ({
    useHistory: () => ({
      push: jest.fn(),
    }),
    useLocation: () => ({
        push: jest.fn(),
      }),
  }));
describe('view research public', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
        <Provider store={store}>
            <ResearchPublic />
        </Provider>, div);
    });
    it('renders correctly', () => {
        const tree = renderer
          .create(
          <Provider store={store}>
            <ResearchPublic />
          </Provider>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
});