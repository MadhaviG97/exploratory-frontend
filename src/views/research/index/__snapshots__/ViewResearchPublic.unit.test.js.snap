// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view research public renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-110"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="MuiBox-root MuiBox-root-111"
  >
    <div>
      <main
        className="MuiGrid-root makeStyles-root-4 MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
        />
        <paper
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 MuiGrid-grid-md-2"
          elevation={6}
          square={true}
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <circle
                color="#014F82"
                size={150}
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <circle
                color="#014F82"
                size={150}
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <circle
                color="#014F82"
                size={150}
              />
            </div>
          </div>
        </paper>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
        />
      </main>
      <div
        className="MuiBox-root MuiBox-root-112"
      />
    </div>
  </div>
</div>
`;
